common/tools.py

```diff
@@ -369,6 +371,162 @@ def get_native_language_and_completeness(language_code):
 
     return (name, completeness)
 
+# |---------------------------------------|
+# | Snapshot handling                     |
+# |                                       |
+# | Candidates for refactoring and moving |
+# | into better suited modules/classes    |
+# |---------------------------------------|
+
+
+def validate_snapshots_path(path: Union[str, pathlib.Path],
+                            host_user_profile: tuple[str, str, str],
+                            mode: str,
+                            copy_links: bool,
+                            error_handler: callable) -> bool:
+    """Check if the given path is valid for being a snapshot path.
+
+    It is checked if it is a folder, if it is writable, if the filesystem is
+    supported and several other things.
+
+    Dev note  (buhtz, 2024-09): That code is a good candidate to get moved
+        into a class or module.
+
+    Args:
+        path: The path to validate as a snapshot path.
+        host_user_profile: I three item list.
+        mode: The profiles mode.
+        copy_links: The copy links value.
+        error_handler: Handle function receiving error messages.
+
+    Returns: Success (`True`) or failure (`False`).
+    """
+    if not isinstance(path, pathlib.Path):
+        path = pathlib.Path(path)
+
+    if not path.is_dir():
+        error_handler(_('Invalid option. {path} is not a folder.')
+                      .format(path=path))
+        return False
+
+    # build full path
+    # <path>/backintime/<host>/<user>/<profile_id>
+    host_user_profile = pathlib.Path(os.sep.join(host_user_profile))
+    full_path = path / 'backintime' / host_user_profile
+
+    # create full_path
+    try:
+        full_path.mkdir(mode=0o777, parents=True, exist_ok=True)
```

Comment:
Should creating the directory happen in a different function?
Or should this function be renamed to reflect the fact that it creates a directory?